### ISSUE TITLE
SingletonDemo

### ISSUE THREAD

**Issue by gargvk81** [dhruvsinghal007/main#4]
@dhruvsinghal007 
Did you read anything about Singletons online as well?

Check what is Double check locking (DCL) and why it's needed. Then fix your SingletonClass and reupload.

**Comment by dhruvesingal** [comment]
I read about DCL. I learnt that lazy initialization can either be safely done through DCL or using volatile keyword. I have modified the project.

**Comment by gargvk81** [comment]
@dhruvsinghal007 
Wrong. It can not be done using volatile. Be very careful while using volatile. avoid it at any cost if possible. 
you have to use DCL.
read here [here](http://stackoverflow.com/questions/15077910/singleton-with-volatile-in-java)

Fix and reupload.

**Comment by dhruvesingal** [comment]
ok...

**Comment by dhruvesingal** [comment]
I already did the DCL in the code. Is it fine?

**Comment by dhruvesingal** [comment]
I mean I reuploaded after adding DCL.

**Comment by dhruvesingal** [comment]
I also read about the locking strategies you mentioned in the link you sent before. It mentioned that we should use DCL and also make the instance volatile at the same time. So I modified and reuploaded.
